idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -1536,3 +1535,199 @@ def sample_points(self):
+
+            Exception: When the **number_of_samples** is invalid (not an integer, too large, zero, negative)
+
+
+        """
+        if sampling_type is None:
+            sampling_type = "creation"
+            self.sampling_type = sampling_type
+            print("Creation-type sampling will be used.")
+        elif not isinstance(sampling_type, str):
+            raise Exception("Invalid sampling type entry. Must be of type <str>.")
+        elif (sampling_type.lower() == "creation") or (
+            sampling_type.lower() == "selection"
+        ):
+            sampling_type = sampling_type.lower()
+            self.sampling_type = sampling_type
+        else:
+            raise Exception(
+                'Invalid sampling type requirement entered. Enter "creation" for sampling from a range or "selection" for selecting samples from a dataset.'
+            )
+        print("Sampling type: ", self.sampling_type, "\n")
+
+        if self.sampling_type == "selection":
+            if isinstance(data_input, (pd.DataFrame, np.ndarray)):
+                self.selection_columns_preprocessing(data_input, xlabels, ylabels)
+            else:
+                raise ValueError(
+                    'Pandas dataframe or numpy array required for sampling_type "selection."'
+                )
+
+            # Catch potential errors in number_of_samples
+            if number_of_samples is None:
+                print(
+                    "\nNo entry for number of samples to be generated. The default value of 5 will be used."
+                )
+                number_of_samples = 5
+            elif number_of_samples > self.data.shape[0]:
+                raise Exception(
+                    "Sample size cannot be greater than number of samples in the input data set"
+                )
+            elif not isinstance(number_of_samples, int):
+                raise Exception("number_of_samples must be an integer.")
+            elif number_of_samples <= 0:
+                raise Exception("number_of_samples must a positive, non-zero integer.")
+            self.number_of_samples = number_of_samples
+
+        elif self.sampling_type == "creation":
+            if not isinstance(data_input, list):
+                raise ValueError(
+                    'List entry of two elements expected for sampling_type "creation."'
+                )
+            elif len(data_input) != 2:
+                raise Exception("data_input must contain two lists of equal lengths.")
+            elif not isinstance(data_input[0], list) or not isinstance(
+                data_input[1], list
+            ):
+                raise Exception("data_input must contain two lists of equal lengths.")
+            elif len(data_input[0]) != len(data_input[1]):
+                raise Exception("data_input must contain two lists of equal lengths.")
+            elif data_input[0] == data_input[1]:
+                raise Exception("Invalid entry: both lists are equal.")
```

Comment:
I think this misses the case where any two matching pairs are equal. I.e. it only checks if ALL dimensions have lower bounds equal to upper bounds, not that any dimension has matching bounds.

Reply:
Fixed, with tests.